packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -336,7 +336,7 @@ class RunCommand extends RunCommandBase {
 
     if (hotMode) {
       for (Device device in devices) {
-        if (!device.supportsHotMode)
+        if (!device.supportsHotReload)
```

Comment:
`!device.supportsHotReload && !device.supportsHotRestart`?

Reply:
Either that or adjust the error message

Reply:
We definitely still want to support hot mode for fuchsia devices, just not restart.  Updated error message